cpp/kiss_icp/CMakeLists.txt

```diff
@@ -33,6 +33,10 @@ set(CMAKE_BUILD_TYPE Release)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
What about using generator expressions? I'm just asking, not requiring to change

```cmake
  add_compile_options("$<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Wall>")
  add_compile_options("$<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Wextra>")
  add_compile_options("$<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wall>")
  add_compile_options("$<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wextra>")
  add_compile_options("$<$<COMPILE_LANG_AND_ID:CXX,GNU>:-pedantic>")
```

Reply:
I'd also add windows here: ` $<$<COMPILE_LANG_AND_ID:CXX,MSVC>:/W4 /WX >`

Reply:
I noticed you have a function ` set_global_target_properties`  to set flags. I understand that this is the "modern" CMake way to do it. Since there's always the risk of forgetting to call this, I'd prefer to set stuff like that globally for the entire project. But in this case, where you have it already, `CompilerOptions.cmake`  is definitely the place for this, I moved it there. Also added `MSVC`.

Reply:
Great, I also agree on that globally might look safer on the surface.

But the key idea behind "Modern cmake", something I like to call "target oriented cmake" is that each target is well defined without contaminating other ones

As a real world example, if you would like to include this project in a (for example) ROS context, and you would 'add_subdirectory(kiss_icp)', then nothing else would compile. Since now the 'Werror" would be propagated to all targets in the build system

Might seem like a corner case, but believe, there is nothing more annoying than working with projects like this 😉